tool/formula/src/main/java/org/openscience/cdk/tools/manipulator/MolecularFormulaManipulator.java

```diff
@@ -1390,4 +1473,89 @@ else if (proton != null &&
 
         return true;
     }
+
+    /**
+     * Helper method for adding isotope distributions to a MF. The method adds
```

Comment:
I don't think I quite understand what this methods is doing... with 'isotope distribution' I understand the relative abundances of the isotopes of one element... but this is something else, right?

Reply:
Well there is the argument if it's not clear it might need to be written/named better.

The gist of this comes from Roger's talk at the RDKit UGM: https://www.nextmovesoftware.com/talks/Sayle_SimpleCheminformaticsProblems_RDKITUGM_201809.pdf where he talks about computing mass and binomial coefficients.

In Roger's example he just shows 2 isotopes.. for example imagine 13C and 14C. To handle the case where there is more than 2 stable isotopes we can think about the problem as follows

```
{most abundant} 1
{2nd most abundant} 2
{3rd most abundant} 3
{4th most abundant} 4
```

first work out the most abundant, which we can think of as partition it as follows: {1}, {2 3 4}. Then we recurse and work out {2} {3 4}, then {3} {4}, etc.

Reply:
We can also compute the relative abundance of each formula distribution using binomials... i.e. isotope generator but the numbers grow very quickly so I didn't add there here.

Reply:
Ah, but then it is about the same thing... Just have to get my head around then how (and why) this partitioning is used in the algorithm (to calculate the relative abundances). Thanks for the extra detail!